src/client/components/Autocomplete/Autocomplete.tsx

```diff
@@ -0,0 +1,67 @@
+import './Autocomplete.scss'
+import React from 'react'
+
+import axios from 'axios'
+import Turnstone from 'turnstone'
+
+type Props = {
+  // styles: Record of autocomplete class names
+  styles?: Record<string, string>
+  // listbox: Object containing autocomplete options,
+  // mainly used for selecting correct key for displayName for object
+  listbox: Record<string, unknown>
+  // url: Should support query, limit params
+  url: string
+  disabled: boolean
+
+  value: string
+  name?: string
+  onChange: (value: any) => void
+}
+
+const Autocomplete = (props: Props) => {
+  const { value, disabled, onChange, listbox, url, styles, name } = props
+
+  if (disabled) return <div className="text-input__readonly-view ">{value}</div>
+
+  // Set up listbox contents.
+  const defaultListbox = {
+    // should return array with { [listbox.displayField]: value }
+    data: (query: string) => axios.get(`${url}?query=${encodeURIComponent(query)}&limit=10`).then(({ data }) => data),
+  }
```

Comment:
this should not be added here but should be passed by the caller as prop.
we can use this autocomplete also from local data, not necessarily Ajax

Reply:
plus, use limit 20 from taxon